From: Modestas Vainius <modax@debian.org>
Subject: sanitize SQLite library version check
Forwarded: no
Last-Update: 2010-08-13

A weird SQLite library version check makes akonadi crash with SQLite 3.7.0 or
later. This patch loosens it a bit. However, even such a variant of the check
does not make much sense because it's basically set to successed or fail at
compiled time. Hence if it is needed, it should be done at cmake/configure
time.
--- a/qsqlite/src/qsql_sqlite.cpp
+++ b/qsqlite/src/qsql_sqlite.cpp
@@ -323,7 +323,7 @@ bool QSQLiteResult::prepare(const QStrin
 
     setSelect(false);
 
-    Q_ASSERT(SQLITE_VERSION_NUMBER == 3006023);
+    Q_ASSERT(SQLITE_VERSION_NUMBER >= 3006023);
 #if (SQLITE_VERSION_NUMBER >= 3003011)
 //    int res = sqlite3_prepare16_v2(d->access, query.constData(), (query.size() + 1) * sizeof(QChar),
 //                                   &d->stmt, 0);
